### ISSUE TITLE
Want to extend detection categories without compromising the original model's functionality.

### ISSUE THREAD

**Issue by wutian-dev** [ultralytics/ultralytics#21180]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I'd like to ask a question. After I finished training a dataset and generated the YAML file, I found that when using the model to recognize images, only this trained model works independently. My original YOLOv8 model becomes unusable. For example, I trained a model to recognize manhole covers, but this model can only identify manhole covers. When I want to recognize images containing different objects, other objects cannot be detected. Meanwhile, when using the original YOLOv8 model, it cannot recognize manhole covers. How can I add more recognition models to the existing YOLOv8 model, so that my YOLOv8 model can detect more types of objects? All my operations are done in Python;

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wutian-dev, thank you for your interest in Ultralytics 🚀! For expanding detection categories or combining custom and original model capabilities, we recommend checking our [Docs](https://docs.ultralytics.com/) for guidance and reviewing the many [Python](https://docs.ultralytics.com/usage/python/) examples available. Many users have similar questions, so the documentation and community forums might already have helpful insights.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much detail as possible—such as your dataset YAML, training logs, and sample images—and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community in the way that works best for you! For real-time chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, see [Discourse](https://community.ultralytics.com/). You can also browse threads and share experiences on our [Subreddit](https://www.reddit.com/r/Ultralytics/).

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response 🤖. An Ultralytics engineer will review your question and assist you further soon!

**Comment by Y-T-G** [comment]
That's how models work. Unless you are training on a dataset of all the classes you want to keep, it will forget the previous classes.

**Comment by wutian-dev** [comment]
> 这就是模型的工作原理。除非你在要保留的所有类的数据集上进行训练，否则它会忘记前面的类。

Can I add new object classes to a pre-trained YOLOv8 model, or do I have to train a new model on a larger dataset in order to detect more objects?

**Comment by Y-T-G** [comment]
You need to have a dataset with images of all the classes you want to detect.
